src/test/scala/com/earnest/util/aws/s3/InteractionTest.scala

```diff
@@ -10,64 +11,71 @@ import cats.syntax.flatMap._
 import io.circe.{Decoder, Encoder}
 import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
 
+import scala.concurrent.ExecutionContext
+
 object s3Env {
-  val s3DS = (config.getFromEnvironment[IO]() >>= s3.connection.createS3DataSource[IO]).unsafeRunSync()
+  val tp = Executors.newFixedThreadPool(2)
+  val blockingEc = ExecutionContext.fromExecutor(tp)
+  implicit val cs = IO.contextShift(ExecutionContext.global)
+  val s3DS = (config.getFromEnvironment[IO]() >>= (conf => s3.connection.createS3DataSource[IO](conf, blockingEc))).unsafeRunSync()
 }
 
 final class InteractionTest extends FreeSpec with Matchers with BeforeAndAfterAll {
   import s3Env._
 
   override def afterAll() = {
     s3DS.tfm.shutdownNow()
+    tp.shutdownNow()
+    ()
```

Comment:
For my learning, why is `()` necessary here?

Reply:
because ` tp.shutdownNow()` does not return `Unit`/`void`. I make the method return Unit via `()`